File: qcmplus/src/main/java/com/backend/qcmplus/model/Reponse.java
package com.backend.qcmplus.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "Reponse")
@Getter
@Setter
@NoArgsConstructor
public class Reponse implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long idReponse;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_question")
    private Question question;

    private String libelle;

    private Boolean isCorrect;

    @Override
    public String toString() {
        return "Reponse [idReponse=" + idReponse + ", iscorrect=" + isCorrect + ", libelle=" + libelle + "]";
    }

}

File: front/src/app/models/reponse.ts

export class Reponse {
    libelle: string;
    isCorrect: boolean;
    idReponse: any;
    question: any;

  constructor(libelle: string, isCorrect: boolean, idReponse: any) {
    this.libelle = libelle;
    this.isCorrect = isCorrect;
    this.idReponse = idReponse;
  }

}
